File: stu_spring/oauth-demo/src/main/java/com/study/todyproject/oauthdemo/config/SecurityConfig.java
package com.study.todyproject.oauthdemo.config;

import com.study.todyproject.oauthdemo.security.jwt.TokenFilter;
import com.study.todyproject.oauthdemo.security.oauth.CustomOAuthUserService;
import com.study.todyproject.oauthdemo.security.oauth.OAuth2SuccessHandler;
import lombok.RequiredArgsConstructor;
import org.springframework.context.annotation.Bean;
import org.springframework.security.config.annotation.web.builders.HttpSecurity;
import org.springframework.security.config.annotation.web.configuration.EnableWebSecurity;
import org.springframework.security.config.annotation.web.configuration.WebSecurityConfigurerAdapter;
import org.springframework.security.crypto.bcrypt.BCryptPasswordEncoder;
import org.springframework.security.crypto.password.PasswordEncoder;
import org.springframework.security.web.authentication.UsernamePasswordAuthenticationFilter;

@RequiredArgsConstructor
@EnableWebSecurity
public class SecurityConfig extends WebSecurityConfigurerAdapter {

    private final CustomOAuthUserService userService;
    private final OAuth2SuccessHandler oAuth2SuccessHandler;
    private final TokenFilter tokenFilter;

    @Override
    protected void configure(HttpSecurity httpSecurity) throws Exception {
        httpSecurity
                .csrf().disable()
                .formLogin().disable()
                .httpBasic().and()
                .authorizeRequests()
                .antMatchers("/").permitAll()
                .anyRequest().permitAll()
                .and()
                .addFilterBefore(tokenFilter, UsernamePasswordAuthenticationFilter.class)
                .oauth2Login()
                .successHandler(oAuth2SuccessHandler)
                .userInfoEndpoint().userService(userService);
    }

    @Bean
    public PasswordEncoder passwordEncoder() {
        return new BCryptPasswordEncoder();
    }

}

File: stu_spring/kotlin_basic/src/main/kotlin/com/kotlin/test/kotlintest/global/security/SecurityConfig.kt
package com.kotlin.test.kotlintest.global.security

import com.kotlin.test.kotlintest.global.exception.ExceptionHandlerFilter
import com.kotlin.test.kotlintest.global.security.authentication.JwtTokenFilter
import com.kotlin.test.kotlintest.global.security.jwt.JwtTokenProvider
import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.Configuration
import org.springframework.http.HttpMethod
import org.springframework.security.config.annotation.web.builders.HttpSecurity
import org.springframework.security.config.annotation.web.configuration.EnableWebSecurity
import org.springframework.security.config.annotation.web.configuration.WebSecurityConfigurerAdapter
import org.springframework.security.crypto.bcrypt.BCryptPasswordEncoder
import org.springframework.security.crypto.password.PasswordEncoder

@Configuration
@EnableWebSecurity
class SecurityConfig(
    private val jwtTokenProvider: JwtTokenProvider
) : WebSecurityConfigurerAdapter() {

    override fun configure(http: HttpSecurity) {
        val exceptionHandlerFilter = ExceptionHandlerFilter()
        val tokenFilter = JwtTokenFilter(jwtTokenProvider)

        http
            .cors().and()
            .csrf().disable()

            .authorizeRequests()
            .antMatchers(HttpMethod.POST, "/auth").permitAll()
            .antMatchers(HttpMethod.POST, "/user").permitAll()
            .antMatchers(HttpMethod.GET, "/auth").permitAll()
            .antMatchers(HttpMethod.GET, "/posts/list").permitAll()
            .anyRequest().authenticated()
            .and().apply(FilterConfig(tokenFilter, exceptionHandlerFilter))

    }

    @Bean
    fun paswordEncoder(): PasswordEncoder {
        return BCryptPasswordEncoder()
    }

}